pinot-query-planner/src/main/java/org/apache/pinot/query/planner/logical/LiteralHintUtils.java

```diff
@@ -46,13 +46,13 @@ public static String literalMapToHintString(Map<Pair<Integer, Integer>, RexExpre
           e.getValue().getDataType().name(), e.getValue().getValue()));
     }
     // semi-colon is used to separate between encoded literals
-    return "{" + StringUtils.join(literalStrings, ";") + "}";
+    return "{" + StringUtils.join(literalStrings, ";:;") + "}";
```

Comment:
where is this `;:;` used?

Reply:
the problem is b/c in some agg literals it allows stringify parameter type with `;` as the separator. this conflicts with my usage of `;` which indicates the "next literal hint".

Thus i used another 3-character splitter `;:;` to indicate hint-literal boundaries.